adapter/service/virtualmeter/manager.go

```diff
@@ -433,14 +416,14 @@ func (m *manager) normalizeOutLvlSwitchLevel(level int64, serviceAddr string) (f
 // createVirtualServicesForThing creates a virtual meter and numeric meter services' specifications depending on
 // presence of other services. Currently virtual metering is support for following services:
 // - outlvlswitch.Service.
-func (m *manager) createVirtualServicesForThing(t adapter.Thing) (outLvlSwitchService *fimptype.Service, numericMeterService *fimptype.Service) {
+func (m *manager) createVirtualServicesForThing(t adapter.Thing, group string) (outLvlSwitchService *fimptype.Service, numericMeterService *fimptype.Service) {
```

Comment:
This method still does not support multi channel things properly.

You should only process services in the iteration that have the group equal to the provided argument.

Consider example: you have thing with group a and b. Group a has out level switch service, group b has only light sensor service. If I ran your code in this situation I'd have virtual meter service and meter service added to both group a and b, as you return those for first encountered matching service.

Another example is where group a and b have their own respective out level switch services. Then you would add virtual meters to both, but both would be based on the same?